pkg/operator/deploy/deploy.go

```diff
@@ -83,49 +83,74 @@ func New(log *logrus.Entry, env env.Interface, oc *api.OpenShiftCluster, arocli
 	}, nil
 }
 
-func (o *operator) staticResources() ([]kruntime.Object, error) {
-	results := []kruntime.Object{}
-	for _, fileBytes := range utilembed.ReadDirRecursive(embeddedFiles, "staticresources") {
-		obj, _, err := scheme.Codecs.UniversalDeserializer().Decode(fileBytes, nil, nil)
-		if err != nil {
-			return nil, err
-		}
+type deployData struct {
+	Image              string
+	Version            string
+	GitCommit          string
+	IsLocalDevelopment bool
+	HasVersion         bool
+}
 
-		// set the image for the deployments
-		if d, ok := obj.(*appsv1.Deployment); ok {
-			if d.Labels == nil {
-				d.Labels = map[string]string{}
-			}
-			var image string
-
-			if o.oc.Properties.OperatorVersion != "" {
-				image = fmt.Sprintf("%s/aro:%s", o.env.ACRDomain(), o.oc.Properties.OperatorVersion)
-				d.Labels["version"] = o.oc.Properties.OperatorVersion
-			} else {
-				image = o.env.AROOperatorImage()
-				d.Labels["version"] = version.GitCommit
-			}
+func templateManifests(data deployData) ([][]byte, error) {
+	templatesRoot, err := template.ParseFS(embeddedFiles, "staticresources/*.yaml")
+	if err != nil {
+		return nil, err
+	}
+	templatesMaster, err := template.ParseFS(embeddedFiles, "staticresources/master/*")
+	if err != nil {
+		return nil, err
+	}
+	templatesWorker, err := template.ParseFS(embeddedFiles, "staticresources/worker/*")
+	if err != nil {
+		return nil, err
+	}
 
-			for i := range d.Spec.Template.Spec.Containers {
-				d.Spec.Template.Spec.Containers[i].Image = image
+	result := make([][]byte, 0)
+	templatesArray := []*template.Template{templatesMaster, templatesRoot, templatesWorker}
 
-				if o.env.IsLocalDevelopmentMode() {
-					d.Spec.Template.Spec.Containers[i].Env = append(d.Spec.Template.Spec.Containers[i].Env, corev1.EnvVar{
-						Name:  "RP_MODE",
-						Value: "development",
-					})
-				}
+	for _, templates := range templatesArray {
+		for _, v := range templates.Templates() {
+			buff := &bytes.Buffer{}
+			if err := v.Execute(buff, data); err != nil {
+				return nil, err
 			}
+			result = append(result, buff.Bytes())
+			buff.Reset()
 		}
+	}
+	return result, nil
+}
 
-		results = append(results, obj)
+func (o *operator) createObjects() ([]kruntime.Object, error) {
```

Comment:
your first suggestion is only a matter of preference, and I prefer logic is better with my way, since we add when we have the objects, instead of instantiating  zero objects and then assigning value to them.

Reply:
Okay  🙂
IMHO SRP still applies for the "k8sObjects" creation, and short functions are always better than long ones.